metabase/metabase
# Reenable posting Slack subscriptions to private slack channels

Fixes #20867 

### How to test

<ol>
  <li>Go to http://localhost:3000/admin/settings/slack</li>
  <li>Click on "Create Slack app"</li>
  <li>Follow onscreen instructions until you get the OAuth Bot string</li>
  <li>Paste the OAuth Bot string in the Slack config page inside Metabase, select a channel to post to</li>
  <li>
    ⚠️ This is a temporary workaround for a separate issue Noah is looking at: in Slack, visit the channel or channels you wish to send Slack Subscriptions to.
  </li>
  <ol type="a">
    <li>In Slack, go to each channel you wish to post subscriptions to.</li>
    <li>Type `/add` until you see "Add Apps to this channel", and add the Metabase app to each channel
  </ol>
  <li>Visit a dashboard</li>
  <li>Click on the "Subscriptions" icon on the top right. It looks like an ✉️</li>
  <li>Click on "a private channel"</li>
  <li>Fill out the private channel name below the radio button</li>
  <li>Click on "Send to Slack now"</li>
  <li>Check your Slack channel. It should have gotten an update</li>
  <li>Set a subscription, say using "hourly". It should work for private channels.</li>
</ol>

#### Other things in this PR

Broke down a large file, was > 500 loc.
Fixed a couple of propTypes.
